### PR TITLE
Migrate to `bevy` v0.6

### PR DESCRIPTION
**Pull request by mnett82**
This PR mostly deals with breaking changes from migrating to `bevy` version 0.6, as per the [Bevy migration guide](https://bevyengine.org/learn/book/migration-guides/0.5-0.6/).

Some callouts:

* Used `StorageType::SparseSet` for `FlyCam`, since I expect it to only exist approximately for one entity.
* Added `[[example]]` sections to the repository config.

### CODE REVIEW

**Comment by mnett82** [comment]
One important note: I did *not* nudge the version to `0.6`, please let me know if that's desirable.

File: src/lib.rs [hunk 1]
```diff
@@ -26,6 +26,8 @@ impl Default for MovementSettings {
 }
 
 /// Used in queries when you want flycams and not other cameras
+#[derive(Component)]
+#[component(storage = "SparseSet")]
```
Comment (BlackPhlox) [inline src/lib.rs:30]: Callout 1:
```suggestion
```

File: Cargo.toml [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
-bevy = { version = "0.5", default-features = false, features = [ "x11", "wayland" ] }
+bevy = {version = "0.6", default-features = false, features = ["x11", "wayland"]}
+
+[[example]]
+name = "basic"
+path = "examples/basic.rs"
+
+[[example]]
+name = "scroll"
+path = "examples/scroll.rs"
```
Comment (BlackPhlox) [inline Cargo.toml:31]: Callout 2:
```suggestion
```

**Review by BlackPhlox — CHANGES_REQUESTED** [review]

**Comment by mnett82** [comment]
@BlackPhlox Thanks for taking a look at the PR!

I bumped the crate version and reverted the changes to the examples. 

I agree, having a large number of `FlyCam` components and massively adding/removing them both seem like unrealistic scenarios. Given that, my choice of `SparseSet` for storage was with the goal to minimize memory footprint. However, I think I misunderstood the default archetype-based storage.

Before reverting this to default storage, could you confirm that this will not grow linearly with the number of entities? Thanks!

**Comment by BlackPhlox** [comment]
Thanks! I don't think there is any discernable difference in memory footprint between the two storage types, as they both grow linearly with the size of entities, though I've asked in the discord just to be sure. From what I can see the storage types only affect iteration and operation speed.

**Comment by BlackPhlox** [comment]
@maniwani (Joy) [from discord](https://discord.com/channels/691052431525675048/742569353878437978/935500970232791050) was amazing at taking their time to educate me on bevy_ecs, I can highly recommend reading the response. So yes, the memory footprint is mostly the same, and Sparse might even be potentially bigger than the default storage type because of the required relations.

**Comment by mnett82** [comment]
Apologies for the slow progress, have been busy at work the last days.

@BlackPhlox Thanks for digging up the details on Discord, definitely learned a lot looking through it. I've removed the `SparseSet` storage type from the PR.

I also pinged @sburris0 via mail since it doesn't seem that they have been active on Github for a while, but let's see what happens.

**Comment by BlackPhlox** [comment]
Thanks and no worries, I've been in contact with Spencer and I have received ownership on crates.io so I can publish a 0.6 version 🎉

**Review by BlackPhlox — APPROVED** [review]

**Pull request merged**